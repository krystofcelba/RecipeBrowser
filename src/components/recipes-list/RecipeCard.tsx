import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import { Card } from 'react-native-elements';

import { Recipe } from 'src/services/api';

interface Props {
  recipe: Recipe;
  onPress: (recipe: Recipe) => void;
}

class RecipeCard extends React.PureComponent<Props> {
  onPress = () => {
    const { recipe, onPress } = this.props;
    onPress(recipe);
  };

  render() {
    const { recipe } = this.props;
    return (
      <TouchableOpacity onPress={this.onPress}>
        <Card image={{ uri: recipe.image }}>
          <Text style={styles.title}>{recipe.name.toUpperCase()}</Text>
        </Card>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  title: {
    fontSize: 14,
    fontWeight: 'bold',
  },
});

export default RecipeCard;
