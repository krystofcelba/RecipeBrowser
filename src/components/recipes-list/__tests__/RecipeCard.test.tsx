import React from 'react';
import { shallow } from 'enzyme';
import RecipeCard from '../RecipeCard';

const mockedRecipes = require('src/__mockData__/recipes.json');

describe('RecipeCard component', () => {
  const mockedRecipe = mockedRecipes[0];
  it('renders as expected', () => {
    const wrapper = shallow(<RecipeCard testID="recipeCard_0" recipe={mockedRecipe} onPress={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls actions as expected on press', () => {
    const wrapper = shallow(
      <RecipeCard
        testID="recipeCard_0"
        recipe={mockedRecipe}
        onPress={recipe => {
          expect(recipe).toEqual(mockedRecipe);
        }}
      />,
    );
    wrapper.simulate('press');
  });
});
