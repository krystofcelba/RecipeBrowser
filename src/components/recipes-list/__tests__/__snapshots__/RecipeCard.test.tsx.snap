// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard component renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  testID="recipeCard_0"
>
  <Card
    image={
      Object {
        "uri": "https://nutrition-assets.freeletics.com/recipes/large/mustard_deviled_eggs.jpg",
      }
    }
  >
    <SubTitleText>
      Mustard Deviled Eggs
    </SubTitleText>
  </Card>
</TouchableOpacity>
`;
